mcs/class/referencesource/System/services/monitoring/system/diagnosticts/Process.cs

```diff
@@ -1346,16 +1348,34 @@ private enum StreamReadMode
                 machineName = ".";
                 raisedOnExited = false;
 
-                //Don't call close on the Readers and writers
-                //since they might be referenced by somebody else while the 
-                //process is still alive but this method called.
-                standardOutput = null;
+                //Call close on streams if the user never saw them.
+                //A stream in the undefined mode was never fetched by the user.
+                //A stream in the async mode is wrapped on a AsyncStreamReader and we should dispose that instead.
+                //  no way for users to get a hand on a AsyncStreamReader.
+                var tmpIn = standardInput;
```

Comment:
Process is not supposed to be thread safe, so having a race conditions shouldn't be a matter in this case.